### PR TITLE
(#902) New feature: Discovered Tiles

### PR DESCRIPTION
**Pull request by AppleC4T**
- [x] Read and understood [see CONTRIBUTING.md](https://github.com/fossasia/labyrinth/blob/master/CONTRIBUTING.md)
- [x] Included a preview link and screenshot showing after and before the changes.
- [x] Included a description of change below.
- [x] Tried to [squash the commits][squash].
- [x] Added Screenshot to Indicate Location of Change in Game

### Changes done in this Pull Request

- View my work at https://pr-911-fossasia-labyrinth.surge.sh or at https://applec4t.github.io/labyrinth/

- Fixes https://github.com/fossasia/labyrinth/issues/902

### The problem I want to solve or the facility I want to improve is/are

Add a _Discovered Tiles_ Counter that would track how many individual tiles the player has seen and allow to make conditions based on this number (e.g. unlock levels, give badges depending on how many tiles the player has discovered).
### My steps to solve it ###

- Added script to the `player.js` file tracking which tiles the player has seen and whether or not the tile he enters is new
- Added the Discovered Tiles Counter in the Statistics menu that shows the number of discovered tiles
- Added 3 methods for making condition based on the number of discovered tiles 

    1. `player.hasDiscoveredExactly(n)` returns `true` if player has discovered exactly _n_ number of tiles
    2. `player.hasDiscoveredAtLeast(n)` returns `true` if player has discovered _n_ or more tiles
    3. `player.hasDiscoveredLessThan(n)` returns `true` if player has discovered less than _n_  tiles 

### Screenshots, Location of Changes

image

image

image

[squash]: https://stackoverflow.com/questions/5189560/squash-my-last-x-commits-together-using-git

### CODE REVIEW

File: js/player.js [hunk 1]
```diff
@@ -123,4 +141,25 @@ function Player() {
...
+			return false;
+		}
+	};
+	this.hasDiscoveredLessThan = function(x){
+		if (seenTiles.length < x){
+			return true;
+		}else{
+			return false;
+		}
+	};
```
Comment (YashKumarVerma) [inline js/player.js]: I think we can consolidate these 3 functions into one which simply gives the number of tiles visited

Comment (AppleC4T) [reply js/player.js]: Sure thing

**Comment by AppleC4T** [comment]
@YashKumarVerma Done.
image

**Comment by AppleC4T** [comment]
@YashKumarVerma can the PR be merged now?

**Comment by AppleC4T** [comment]
Thanks!

**Pull request merged**